java/core/src/java/org/apache/orc/filter/OrcFilterContext.java

```diff
@@ -0,0 +1,169 @@
+      this.batch = batch;
+      vectors.clear();
+    }
+    return this;
+  }
+
+  @Override
+  public void setFilterContext(boolean selectedInUse, int[] selected, int selectedSize) {
+    batch.setFilterContext(selectedInUse, selected, selectedSize);
+  }
+
+  @Override
+  public boolean validateSelected() {
+    return batch.validateSelected();
+  }
+
+  @Override
+  public int[] updateSelected(int i) {
+    return batch.updateSelected(i);
+  }
+
+  @Override
+  public void setSelectedInUse(boolean b) {
+    batch.setSelectedInUse(b);
+  }
+
+  @Override
+  public void setSelected(int[] ints) {
+    batch.setSelected(ints);
+  }
+
+  @Override
+  public void setSelectedSize(int i) {
+    batch.setSelectedSize(i);
+  }
+
+  @Override
+  public void reset() {
+    batch.reset();
+  }
+
+  @Override
+  public boolean isSelectedInUse() {
+    return batch.isSelectedInUse();
+  }
+
+  @Override
+  public int[] getSelected() {
+    return batch.getSelected();
+  }
+
+  @Override
+  public int getSelectedSize() {
+    return batch.getSelectedSize();
+  }
+
+  public ColumnVector[] getCols() {
+    return batch.cols;
+  }
+
+  /**
+   * Retrieves the column vector that matches the specified name. Allows support for nested struct
+   * references e.g. order.date where data is a field in a struct called order.
+   *
+   * @param name The column name whose vector should be retrieved
+   * @return The column vector
+   * @throws IllegalArgumentException if the field is not found or if the nested field is not part
+   *                                  of a struct
+   */
+  public ColumnVector findColumnVector(String name) {
+    if (!vectors.containsKey(name)) {
+      vectors.put(name, findVector(name));
+    }
+
+    return vectors.get(name);
+  }
+
+  private ColumnVector findVector(String name) {
+    String[] refs = name.split(SPLIT_ON_PERIOD);
+    TypeDescription schema = readSchema;
+    List<String> fNames = schema.getFieldNames();
+    ColumnVector[] cols = batch.cols;
+    ColumnVector vector;
+
+    // Try to find the required column at the top level
+    int idx = findVectorIdx(refs[0], fNames);
+    vector = cols[idx];
+
+    // In case we are asking for a nested search further. Only nested struct references are allowed.
+    for (int i = 1; i < refs.length; i++) {
+      schema = schema.getChildren().get(idx);
+      if (schema.getCategory() != TypeDescription.Category.STRUCT) {
+        throw new IllegalArgumentException(String.format(
+          "Field %s:%s not a struct field does not allow nested reference",
+          refs[i - 1],
+          schema));
+      }
+      cols = ((StructColumnVector) vector).fields;
+      vector = cols[findVectorIdx(refs[i], schema.getFieldNames())];
+    }
+
+    return vector;
+  }
+
+  private static int findVectorIdx(String name, List<String> fieldNames) {
+    int idx = fieldNames.indexOf(name);
+    if (idx < 0) {
+      throw new IllegalArgumentException(String.format("Field %s not found in %s",
```

Comment:
Here the focus is on the read schema. Schema evolution will already map the file schema into the read schema.
So in case the read schema has the column even if the file does not have the column the vector will be found and returned.
The case that is not supported is about find a column absent in the read schema.